app/src/org/commcare/activities/TemplatePrinterActivity.java

```diff
@@ -60,39 +72,114 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_template_printer);
 
+        String printStyle = this.getIntent().getExtras().getString(KEY_TEMPLATE_STYLE);
+        if(printStyle == null) {
+            if(CompoundIntentList.isIntentCompound(this.getIntent())) {
+                //Only zebra print jobs can compound
+                //TODO: This still isn't a particularly great way for us to be differentiating
+                printStyle = TEMPLATE_STYLE_ZPL;
+            } else {
+                printStyle = TEMPLATE_STYLE_HTML;
+            }
+        }
+
+        //Since this and the callout activities are raised as "dialog" activities, they will
+        //recreate themselves on rotation. If we detect that we need to not "re-kick-off" the
+        //activity, it will result in duplicate activities.
+        if(printStyle == TEMPLATE_STYLE_ZPL) {
```

Comment:
`==` -> `.equals` or better, just lift this logic into the place where you set `printStyle` to `TEMPLATE_STYLE_ZPL`